src/main/kotlin/no/nav/syfo/client/graphapi/GraphApiClient.kt

```diff
@@ -16,100 +17,133 @@ import org.slf4j.LoggerFactory
+            try {
+                val queryFilter = "startsWith(onPremisesSamAccountName, '$veilederIdent')"
+                val queryFilterWhitespaceEncoded = queryFilter.replace(" ", "%20")
+                val url =
+                    "$baseUrl/v1.0/users?\$filter=$queryFilterWhitespaceEncoded&\$select=onPremisesSamAccountName,givenName,surname,mail,businessPhones&\$count=true"
 
-            val response: GraphApiGetUserResponse = httpClient.get(url) {
-                header(HttpHeaders.Authorization, bearerHeader(oboToken))
-                header("ConsistencyLevel", "eventual")
-                accept(ContentType.Application.Json)
-            }.body()
-            COUNT_CALL_GRAPHAPI_VEILEDER_SUCCESS.increment()
-            response
-        } catch (e: ResponseException) {
-            COUNT_CALL_GRAPHAPI_VEILEDER_FAIL.increment()
-            log.error(
-                "Error while requesting Veileder from GraphApi {}, {}, {}",
-                StructuredArguments.keyValue("statusCode", e.response.status.value.toString()),
-                StructuredArguments.keyValue("message", e.message),
-                callIdArgument(callId),
-            )
-            throw e
+                val response: GraphApiGetUserResponse = httpClient.get(url) {
+                    header(HttpHeaders.Authorization, bearerHeader(oboToken))
+                    header("ConsistencyLevel", "eventual")
+                    accept(ContentType.Application.Json)
+                }.body()
+                COUNT_CALL_GRAPHAPI_VEILEDER_SUCCESS.increment()
+                val graphAPIUser = response.value.firstOrNull()
+                if (graphAPIUser != null) {
+                    cache.setObject(
+                        expireSeconds = 3600,
+                        key = cacheKey,
+                        value = graphAPIUser,
+                    )
+                }
+                COUNT_CALL_GRAPHAPI_VEILEDER_CACHE_MISS.increment()
+                graphAPIUser
+            } catch (e: ResponseException) {
+                COUNT_CALL_GRAPHAPI_VEILEDER_FAIL.increment()
+                log.error(
+                    "Error while requesting Veileder from GraphApi {}, {}, {}",
+                    StructuredArguments.keyValue("statusCode", e.response.status.value.toString()),
+                    StructuredArguments.keyValue("message", e.message),
+                    callIdArgument(callId),
+                )
+                throw e
+            }
         }
     }
 
     suspend fun veilederList(
+        enhetNr: String,
         axsysVeilederlist: List<AxsysVeileder>,
         callId: String,
         token: String,
     ): List<Veileder> {
-        val oboToken = azureAdClient.getOnBehalfOfToken(
-            scopeClientId = baseUrl,
-            token = token,
-        )?.accessToken ?: throw RuntimeException("Failed to request access to Enhet: Failed to get OBO token")
+        val cacheKey = "$GRAPH_API_CACHE_VEILEDER_LISTE_PREFIX$enhetNr"
```

Comment:
Første forsøk her virker ikke bra nok (det tar for lang tid å få opp lista med veiledere), så forsøker å legge tilbake den gamle batch-koden og i tillegg cache responsen.

Reply:
Hvordan blir egentlig sammenhengen mellom axsysVeilederList og det som caches her? :thinking:

Reply:
Det vil i hovedsak være den samme lista, men lista fra axsys-kallet har bare identer mens lista etter at den er "beriket" med graphApi-data også inkluder navn.